source/app/controllers/urls_controller.rb

```diff
@@ -1,2 +1,42 @@
 class UrlsController < ApplicationController
+
+  def index
+    @urls = Url.all
+  end
+
+  def new
+  end
+
+  def create
+    @url = Url.new(url_params)
+
+    if @url.save
+      redirect_to '/'
+    else
+      @urls = Url.all
+      render 'index'
+    end
+  end
+
+  def destroy
+    @url = Url.find(params[:id])
+    @url.destroy
+
+    redirect_to '/'
+  end
+
+  def click_link
+    @url = Url.find(params[:format])
+
+    new_count = @url.click_count + 1
+    @url.update_attribute(:click_count, new_count)
```

Comment:
There is an `increment` and an `increment!` but isn't the `increment_counter` class method the only one that's atomic? Otherwise, I had misread that in the docs.